hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemAppend.java

```diff
@@ -603,4 +609,94 @@ public void testAppendWithLease() throws Exception {
     assertNotNull(outputStream.getLeaseId());
   }
 
+
+  /**
+   * If a write operation fails asynchronously, when the next write comes once failure is
+   * registered, that operation would fail with the exception caught on previous
+   * write operation.
+   * The next close would also fail for the last caught exception.
+   */
+  @Test
+  public void testIntermittentAppendFailureToBeReported() throws Exception {
```

Comment:
Can this test be broken into smaller tests based on the 3 scenarios listed in the description section ?

Reply:
Taken.